prov/efa/src/efa_domain.c

```diff
@@ -94,6 +107,25 @@ static int efa_open_device_by_name(struct efa_domain *domain, const char *name)
 		}
 	}
 
+	/*
+	 * Check if a PD has already been allocated for this device and reuse
+	 * it if this is the case.
+	 */
+	fastlock_acquire(&pd_list_lock);
```

Comment:
Can you remind me why we need this fastlock?

Reply:
We have to deal with the multi-threaded case where two threads try to open a domain and race to get or release their reference on this global pointer to the pd struct. I'm not sure if this will happen in practice, but it's good to be proactive about it.

Reply:
I see, thanks for your explanation.